es6-shim.js

```diff
@@ -291,6 +291,13 @@
       return x != null && Object(x) === x;
     },
 
+    TypeIsNotObject: function (val) {
+      // This is faster when TypeIsObject may commonly return false.
+      return val === void 0 || val === null || val === true || val === false ||
```

Comment:
From reading
http://comments.gmane.org/gmane.comp.lang.javascript.ecmascript4.general/34503
and http://dhtmlkitchen.com/exotic-objects/ it seems that most exotic
(host) objects actually return "object" or "function".  I couldn't find an
example that returned something else but should still be considered an
object.

Perhaps the easiest thing to do is merge my PR and then do a follow-up
patch which replaces TypeIsObject with your preferred version?  I still
maintain it's not actually correct according to the spec, but I also agree
that adding a huge list of SIMD cases here would be very unpleasant.